### PR TITLE
Return a dataframe from stocks search, removed export to file system (#3923)

### PR DESCRIPTION
**Pull request by joey-walker**
# Description
A small change to allow the search command under stocks to return a dataframe in memory so that you don't have to grab the results from the file system.  Added a flag to configure enabling the exporting of the results to the filesystem.

# How has this been tested?
Ran the terminal to make sure the search command under stocks still works.  Tested with flag both on and off.

Also tested via importing and running under the python console.

Ran the following commands:
## OpenBB Terminal
```shell
 / $ stocks
/stocks/ $ search NBH

                                  Companies found on term nbh                                   
┏━━━━━━┳━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━┓
┃      ┃ Name               ┃ Country       ┃ Sector             ┃ Industry         ┃ Exchange ┃
┡━━━━━━╇━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━┩
│ FNHM │ FNBH Bancorp, Inc. │ United States │ Financial Services │ Banks - Regional │ PNK      │
└──────┴────────────────────┴───────────────┴────────────────────┴──────────────────┴──────────┘

```

## Python
```python
Python 3.8.0 (tags/v3.8.0:fa919fd, Oct 14 2019, 19:37:50) [MSC v.1916 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> from openbb_terminal.sdk import openbb
>>> openbb.stocks.search(country="united states", exchange_country="Germany", limit=5) 
                             Companies found on an exchange in Germany in United States                             
┏━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━┓
┃       ┃ Name                     ┃ Country       ┃ Sector             ┃ Industry                      ┃ Exchange ┃
┡━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━┩
│ 001.F │ Cardtronics plc          │ United States │ Industrials        │ Business Equipment & Supplies │ FRA      │
├───────┼──────────────────────────┼───────────────┼────────────────────┼───────────────────────────────┼──────────┤
│ NWCI  │ NewCardio, Inc.          │ United States │ Healthcare         │ Medical Devices               │ PNK      │
├───────┼──────────────────────────┼───────────────┼────────────────────┼───────────────────────────────┼──────────┤
│ NWE   │ NorthWestern Corporation │ United States │ Utilities          │ Utilities - Diversified       │ NMS      │
├───────┼──────────────────────────┼───────────────┼────────────────────┼───────────────────────────────┼──────────┤
│ NWFL  │ Norwood Financial Corp.  │ United States │ Financial Services │ Banks - Regional              │ NMS      │
├───────┼──────────────────────────┼───────────────┼────────────────────┼───────────────────────────────┼──────────┤
│ NWGC  │ New World Gold Corp.     │ United States │ Basic Materials    │ Gold                          │ PNK      │
└───────┴──────────────────────────┴───────────────┴────────────────────┴───────────────────────────────┴──────────┘
                                                      name        country                  sector                       industry exchange
001.F                                      Cardtronics plc  United States             Industrials  Business Equipment & Supplies      FRA
NWCI                                       NewCardio, Inc.  United States              Healthcare                Medical Devices      PNK
NWE                               NorthWestern Corporation  United States               Utilities        Utilities - Diversified      NMS
NWFL                               Norwood Financial Corp.  United States      Financial Services               Banks - Regional      NMS
NWGC                                  New World Gold Corp.  United States         Basic Materials                           Gold      PNK
...                                                    ...            ...                     ...                            ...      ...
EO5.F                                  EOG Resources, Inc.  United States                  Energy                  Oil & Gas E&P      FRA
EOB.F                                         Arcosa, Inc.  United States             Industrials      Infrastructure Operations      FRA
EOD      Wells Fargo Advantage Funds - Wells Fargo Glob...  United States      Financial Services               Asset Management      NYQ
ENPH.MX                               Enphase Energy, Inc.  United States              Technology                          Solar      MEX
ZZA.F                              Cinemark Holdings, Inc.  United States  Communication Services                  Entertainment      FRA

[14503 rows x 5 columns]

>>> df=openbb.stocks.search(country="united states", exchange_country="Germany", limit=5, export_to_file=False) 
                             Companies found on an exchange in Germany in United States                             
┏━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━┓
┃       ┃ Name                     ┃ Country       ┃ Sector             ┃ Industry                      ┃ Exchange ┃
┡━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━┩
│ 001.F │ Cardtronics plc          │ United States │ Industrials        │ Business Equipment & Supplies │ FRA      │
├───────┼──────────────────────────┼───────────────┼────────────────────┼───────────────────────────────┼──────────┤
│ NWCI  │ NewCardio, Inc.          │ United States │ Healthcare         │ Medical Devices               │ PNK      │
├───────┼──────────────────────────┼───────────────┼────────────────────┼───────────────────────────────┼──────────┤
│ NWE   │ NorthWestern Corporation │ United States │ Utilities          │ Utilities - Diversified       │ NMS      │
├───────┼──────────────────────────┼───────────────┼────────────────────┼───────────────────────────────┼──────────┤
│ NWFL  │ Norwood Financial Corp.  │ United States │ Financial Services │ Banks - Regional              │ NMS      │
├───────┼──────────────────────────┼───────────────┼────────────────────┼───────────────────────────────┼──────────┤
│ NWGC  │ New World Gold Corp.     │ United States │ Basic Materials    │ Gold                          │ PNK      │
└───────┴──────────────────────────┴───────────────┴────────────────────┴───────────────────────────────┴──────────┘

>>> df.head()
                           name        country              sector                       industry exchange
001.F           Cardtronics plc  United States         Industrials  Business Equipment & Supplies      FRA
NWCI            NewCardio, Inc.  United States          Healthcare                Medical Devices      PNK
NWE    NorthWestern Corporation  United States           Utilities        Utilities - Diversified      NMS
NWFL    Norwood Financial Corp.  United States  Financial Services               Banks - Regional      NMS
NWGC       New World Gold Corp.  United States     Basic Materials                           Gold      PNK

```

### CODE REVIEW

File: openbb_terminal/stocks/stocks_helper.py [hunk 1]
```diff
@@ -227,14 +230,16 @@ def search(
...
         title=title,
     )
+    if export_to_file:
+        export_data(
+            export,
+            os.path.dirname(os.path.abspath(__file__)),
+            "search",
+            df,
+            sheet_name,
+        )
```
Comment (jmaslek) [inline openbb_terminal/stocks/stocks_helper.py:240]: I am not following the logic of this `export_to_file`.  If export=="", which it is by default, nothing gets saved.

The way we have our sdk is that exporting should not be handled through this.  Returning a df allows you to save, ie `openbb.stocks.search("Apple").to_csv("path/to/file"`

Comment (joey-walker) [reply openbb_terminal/stocks/stocks_helper.py:240]: Ah, you are right.  The underlying function does have that check, I should have noticed.  

I am not sure if I fully understand your second point as I see that as feature/benefit.  Is it going against the design methodology?  Is there an alternative method for retrieving the results into memory without hitting the file system?

Comment (jmaslek) [reply openbb_terminal/stocks/stocks_helper.py:240]: So it is against our design to have the export in the sdk model functions.  In this example, if you just call

df = openbb.stocks.search("Apple")

That will be in your memory locally.  If you want to save to a file you can using pandas built-ins.  

I might not be understanding what you wish to accomplish.  When you say you want it into memory, what are you hoping to do?

Comment (joey-walker) [reply openbb_terminal/stocks/stocks_helper.py]: I just pushed some commits to remove that redundant flag that I added.  Updated documentation as well.

The desire behind this pull request is to enable this particular function to actually return the results in dataframe format - if any are available -  exactly as you said: `df = openbb.stocks.search("Apple")`

Right now - from my understanding - It doesn't return anything.

Comment (jmaslek) [reply openbb_terminal/stocks/stocks_helper.py]: Yes that is correct.  You fixed that by returning the df.  We just dont need to have the export/sheet name in the model functions.

Comment (joey-walker) [reply openbb_terminal/stocks/stocks_helper.py]: Ah now I understand! (Hopefully).  

I pushed a commit that removes the export/sheet name in that function.  I adjusted and ran the tests to success.

Also redid my initial manual testing.

I also saw the mypy failures, and changed the return statements to give back an empty dataframe.   Similar to other functions in the same file.

**Review by jmaslek — APPROVED** [review]

**Pull request merged**